presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestEliminateCrossJoins.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder;
+import com.facebook.presto.sql.planner.iterative.rule.test.RuleTester;
+import com.facebook.presto.sql.planner.plan.JoinNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.SystemSessionProperties.REORDER_JOINS;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.any;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.join;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.project;
+
+public class TestEliminateCrossJoins
+{
+    private final RuleTester tester = new RuleTester(ImmutableMap.of(
+            REORDER_JOINS, "true"
+    ));
+
+    @Test
+    public void testEliminateCrossJoin()
+    {
+        tester.assertThat(new EliminateCrossJoins())
+                .on(this::joinAndCrossJoin)
+                .matches(
+                        project(
+                                join(JoinNode.Type.INNER,
+                                        ImmutableList.of(aliases -> new JoinNode.EquiJoinClause(new Symbol("cySymbol"), new Symbol("bySymbol"))),
```

Comment:
`join(..)` does not take a list of `EquiJoinClause`-s, it takes list of factories of `EquiJoinClause`. We have a DSL method for the latter (`com.facebook.presto.sql.planner.assertions.PlanMatchPattern#equiJoinClause`), but this case is a bit special and I cannot use that.